### PR TITLE
corrected DAE, modify pom

### LINKED ISSUE #1034: [Bug]: Fix BodyInterceptor: Dead Assignment Eliminator
### What happened?

doesn't remove the dead assignments. For example: x = 10; is a dead assignment, as well as x = 20; since they are reassigned to new values or not used after that statement.
`int tc6_1(int temp) {
        int x = 10;
        x = 30;
        temp = x;
        if (temp > 5) {
            x = someMethod();
        }
        x = 20;
        return temp;
    }`
    
current Output:
   'this := @this: sootUp.RQ1.jb_dae.JB_DAE;
temp := @parameter0: int;
x = 10;
x = 30;
temp = x;

if temp <= 5 goto label1;
x = virtualinvoke this.<sootUp.RQ1.jb_dae.JB_DAE: int someMethod()>();

label1:
x = 20;

return temp;'

Expected Output:
`this := @this: sootUp.RQ1.jb_dae.JB_DAE;
temp := @parameter0: int;
x = 30;
temp = x;

if temp <= 5 goto label1;
x = virtualinvoke this.<sootUp.RQ1.jb_dae.JB_DAE: int someMethod()>();

label1:
return temp;`

### Version

Latest develop branch

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by sahilagichani14**

### CODE REVIEW

**Comment by sahilagichani14** [comment]
WhatsApp Image 2024-09-13 at 19 59 04

File: pom.xml [hunk 1]
```diff
@@ -446,7 +447,7 @@
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-simple</artifactId>
-            <version>2.0.13</version>
+            <version>${slf4j-simple.version}</version>
```
Comment (swissiety) [inline pom.xml]: dependency mgmt

**Review by swissiety — COMMENTED** [review]
and more dep mgmt to remove

File: pom.xml [hunk 2]
```diff
@@ -458,7 +464,7 @@
         <dependency>
             <groupId>org.mockito</groupId>
             <artifactId>mockito-junit-jupiter</artifactId>
-            <version>5.12.0</version>
+            <version>4.0.0</version>
```
Comment (swissiety) [inline pom.xml]: lets make use of dependency mgmt here as well

**Review by swissiety — APPROVED** [review]

File: pom.xml (hunk 2 above)
Comment (sahilagichani14) [reply pom.xml]: so should I add v4 in apk.parser and revert back to v5 here right? @swissiety

**Pull request merged**